packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts

```diff
@@ -277,7 +278,7 @@ export class AdapterDayjs implements MuiPickersAdapter<Dayjs, string> {
     if (timezone !== 'UTC') {
       const fixedValue = value.tz(this.cleanTimezone(timezone), true);
       // @ts-ignore
-      if ((fixedValue.$offset ?? 0) === (value.$offset ?? 0)) {
+      if (fixedValue.$offset === value.$offset) {
```

Comment:
Are we sure `$offset` is always defined?

Reply:
Initially, I thought that it can't when the timezone plugin is defined, but in reality—it can, because you can still create objects with `dayjs` instead of `dayjs.tz`. 🙈 

Do we agree that `fixedOffset` can never have `undefined` `$offset`, because there is `tz` call?
In such a case, I'll try adding a test to cover the case when `value.$offset` is undefined.

Reply:
I reverted the to keep `value.$offset ?? 0` and it looks like existing tests are enough to cover this case. 👍 
So, my assumption somewhat checks out, that `fixedValue.$offset` can never be undefined. 🤔 🤷

Reply:
Make sense, thanks for the double check :+1: